python/tvm/script/builder/tir/block_frame.py

```diff
@@ -46,15 +46,19 @@ def where(predicate) -> None:
     _ffi_api.Where(predicate)  # pylint: disable=no-member # type: ignore
 
 
-def reads(buffer_slices: List[Union[BufferRegion, BufferLoad]]) -> None:
-    if not isinstance(buffer_slices, List):
-        buffer_slices = [buffer_slices]
+def reads(*buffer_slices: List[Union[BufferRegion, BufferLoad]]) -> None:
+    if isinstance(buffer_slices, tuple):
+        buffer_slices = list(buffer_slices)
+    elif len(buffer_slices) == 1 and not isinstance(buffer_slices, list):
+        buffer_slices = [buffer_slices[0]]
```

Comment:
sorry i still dont get the idea here...for example,

```python
T.reads((A[...], ))
```

which could make it go wrong. doesn't it?